custom/icds_reports/utils/aggregation_helpers/distributed/agg_bihar_api_demographics.py

```diff
@@ -0,0 +1,132 @@
+            ('district_id', 'person_list.district_id'),
+            ('block_id', 'person_list.block_id'),
+            ('supervisor_id', 'person_list.supervisor_id'),
+            ('awc_id', 'person_list.awc_id'),
+            ('month', f"'{month_start_string}'"),
+            ('person_id', 'person_list.doc_id'),
+            ('person_name', 'person_list.name'),
+            ('has_adhaar', 'CASE WHEN person_list.aadhar_date is not null THEN 1 ELSE 0 END'),
+            ('bank_account_number', 'person_list.bank_account_number'),
+            ('ifsc_code', 'person_list.ifsc_code'),
+            ('age_at_reg', 'person_list.age_at_reg'),
+            ('dob', 'person_list.dob'),
+            ('gender', 'person_list.sex'),
+            ('blood_group', 'person_list.blood_group'),
+            ('disabled', 'person_list.disabled'),
+            ('disability_type', 'person_list.disability_type'),
+            ('referral_status', 'person_list.referral_status'),
+            ('migration_status', 'migration_tab.is_migrated'),
+            ('resident', 'person_list.resident'),
+            ('registered_status', 'person_list.registered_status'),
+            ('rch_id', 'person_list.rch_id'),
+            ('mcts_id', 'person_list.mcts_id'),
+            ('phone_number', 'person_list.phone_number'),
+            ('date_death', 'person_list.date_death'),
+            ('site_death', 'person_list.site_death'),
+            ('closed_on', 'person_list.closed_on'),
+            ('reason_closure', 'person_list.reason_closure'),
+            ('household_id', 'hh_list.doc_id'),
+            ('household_name', 'hh_list.name'),
+            ('hh_reg_date', 'hh_list.hh_reg_date'),
+            ('hh_num', 'hh_list.hh_num'),
+            ('hh_gps_location', 'hh_list.hh_gps_location'),
+            ('hh_caste', 'hh_list.hh_caste'),
+            ('hh_bpl_apl', 'hh_list.hh_bpl_apl'),
+            ('hh_minority', 'hh_list.hh_minority'),
+            ('hh_religion', 'hh_list.hh_religion'),
+
+        )
+        column_names = ", ".join([col[0] for col in columns])
+        calculations = ", ".join([col[1] for col in columns])
+
+        return f"""
+                INSERT INTO "{self.monthly_tablename}" (
+                    {column_names}
+                )
+                (
+                SELECT
+                {calculations}
+                from "{person_case_ucr}" person_list
+                LEFT JOIN "{AGG_MIGRATION_TABLE}" migration_tab ON (
+                    person_list.doc_id = migration_tab.person_case_id AND
+                    person_list.supervisor_id = migration_tab.supervisor_id AND
+                    migration_tab.month='{month_start_string}'
+                )
+                LEFT JOIN "{household_ucr}" hh_list ON (
+                    person_list.household_case_id = hh_list.doc_id AND
+                    person_list.supervisor_id = hh_list.supervisor_id
+                )
+                WHERE (person_list.opened_on <= '{month_end_string}' AND
```

Comment:
@calellowitz  Since this is Just for Bihar. I think we can first restrict the aggregation for only one state for now because anyway no other state is going to use it atleast in the near future. 
making it to only Bihar will give us sense of its size/speed and it will also reduce chance of reiteration of it because of slow speed(in case it becomes not as fast as they want).

Reply:
Are you planning to only restrict to one state? That would definitely be a start

Reply:
Yeah, I really think it would make our life easy for this one. was putting out here so that you could see any concerns if any.